### ISSUE TITLE
.loc behavior when using 'slice'

### ISSUE THREAD

**Issue by beobest2** [databricks/koalas#1158]
I test as shown in the documentation below. 
I found something strange.

https://koalas.readthedocs.io/en/latest/reference/api/databricks.koalas.Series.loc.html
![image](https://user-images.githubusercontent.com/7010554/71557962-02927a80-2a91-11ea-94e1-c448125d5a37.png)

pandas :
'sidewinder' is at the end of the list, so it prints normally.

```python
>>> pdf = pd.DataFrame([[1, 2], [4, 5], [7, 8]], 
...             index=['cobra', 'viper', 'sidewinder'],  
...             columns=['max_speed', 'shield'])
>>> pdf.loc['cobra':'viper', 'max_speed']
cobra    1
viper    4
Name: max_speed, dtype: int64

```

but koalas prints 'sidewinder'

```python
>>> kdf = ks.DataFrame([[1, 2], [4, 5], [7, 8]],
...              index=['cobra', 'viper', 'sidewinder'],
...              columns=['max_speed', 'shield'])
>>> kdf.loc['cobra':'viper', 'max_speed']
cobra         1
viper         4
sidewinder    7
Name: max_speed, dtype: int64

>>> kdf.loc['cobra':, 'max_speed']
cobra         1
viper         4
sidewinder    7
Name: max_speed, dtype: int64

>>> kdf.loc['sidewinder':, 'max_speed']
viper         4
sidewinder    7
Name: max_speed, dtype: int64

>>> kdf.loc['viper':, 'max_speed']
viper    4
Name: max_speed, dtype: int64
```

Input is done in the order ['cobra', 'viper', 'sidewinder'], 
but koals appears to be recognized as ['cobra', 'sidewinder', 'viper'].

**Comment by itholic** [comment]
thanks for reporting, let me take a look at this

**Comment by itholic** [comment]
fix this at #1159 

## summary

the existing logic was determining range from start to stop by string order alphabetically, so couldn't keep natural order.

we have to determine the range from start to stop with natural order like pandas.

## for more detail

for example, let's assume that we have `kdf` like below

```python
>>> kdf
            max_speed  shield
cobra               1       2
viper               4       5
sidewinder          7       8
```

and perform `iloc` on this with `slice` from 'cobra' to 'viper' like below,

```python
>>> kdf.loc['cobra':'viper', 'max_speed']
```

and we expect that the result will not include 'sidewinder' since it is really not between 'cobra' and 'viper' shown `kdf` above.
(actually, more fundamentally, because pandas is doing like this)

so i fixed it with our new feature `NATURAL_ORDER_COLUMN_NAME` to keep natural order.
